src/main/java/com/neokortex/task/Task.java

```diff
@@ -0,0 +1,80 @@
+package com.elsria.task;
+
+import com.elsria.exceptions.InvalidTaskSerializationException;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+public abstract class Task {
+    private static final char markedSymbol = 'X';
+    private static final char unmarkedSymbol = ' ';
+    protected String description;
+    protected boolean isMarked;
+
+    public Task(String description) {
+        this.description = description;
+        this.isMarked = false;
+    }
+
+    public Task(String description, boolean isMarked) {
+        this.description = description;
+        this.isMarked = isMarked;
+    }
+
+    public void mark() {
+        this.isMarked = true;
+    }
+
+    public void unmark() {
+        this.isMarked = false;
+    }
+
+    public boolean isMarked() {
+        return this.isMarked;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public abstract char taskType();
+
+    public boolean containsKeyword(String keyword) {
+        return this.description.toLowerCase().contains(keyword.toLowerCase());
+    }
+
+    @Override
+    public String toString() {
+        char markStatus = this.isMarked ? markedSymbol : unmarkedSymbol;
+        return String.format("[%c][%c] %s", this.taskType(), markStatus, description);
```

Comment:
Suggestion: One formatter
Keep display patterns in one place to avoid drift.
```suggestion: e.g., new util class
public final class DateFormats {
    public static final DateTimeFormatter DISPLAY = DateTimeFormatter.ofPattern("MMM d yyyy");
    private DateFormats() {}
}
```